### PR TITLE
feat(octoeverywhere): implement update function

### PR DESCRIPTION
**Pull request by th33xitus**

### CODE REVIEW

File: scripts/octoeverywhere.sh [hunk 1]
```diff
@@ -273,9 +273,51 @@ function remove_octoeverywhere()
 #===================================================#
 
 function update_octoeverywhere() {
-  # Since our update might require new python packages or system packages, ask the user to use the
-  # moonraker update manager to do it, since that takes care of it.
-  print_error "Please use the Moonraker Update Manager from the Mainsail or Fluidd UI to update OctoEverywhere.\n Contact our support team if you need help! support@octoeverywhere.com"
+  do_action_service "stop" "octoeverywhere"
```
Comment (QuinnDamerell) [inline scripts/octoeverywhere.sh:276]: Will this handle the octoeveryhere-* other possible service names?

File: scripts/octoeverywhere.sh [hunk 2]
```diff
@@ -273,9 +273,51 @@ function remove_octoeverywhere()
...
+
+  if [[ ! -d ${OCTOEVERYWHERE_DIR} ]]; then
+    clone_moonraker "${OCTOEVERYWHERE_DIR}"
```
Comment (QuinnDamerell) [inline scripts/octoeverywhere.sh:279]: I assume this name was just never updated for generic git clone, and it's not oversight?

File: scripts/octoeverywhere.sh [hunk 3]
```diff
@@ -273,9 +273,51 @@ function remove_octoeverywhere()
...
+  else
+    backup_before_update "octoeverywhere"
+    status_msg "Updating OctoEverywhere for Klipper ..."
+    cd "${OCTOEVERYWHERE_DIR}" && git pull
+    ### read PKGLIST and install possible new dependencies
+    install_octoeverywhere_dependencies
+    ### install possible new python dependencies
+    "${OCTOEVERYWHERE_ENV}"/bin/pip install -r "${OCTOEVERYWHERE_DIR}/requirements.txt"
```
Comment (QuinnDamerell) [inline scripts/octoeverywhere.sh:287]: I don't know much about virtual environments, but I assume this installs into it? The only way I have interactive with virtual envs is by using the /bin/activate the script.

File: scripts/octoeverywhere.sh [hunk 4]
```diff
@@ -273,9 +273,51 @@ function remove_octoeverywhere()
...
+  fi
+
+  ok_msg "Update complete!"
+  do_action_service "restart" "octoeverywhere"
+}
+
+function install_octoeverywhere_dependencies() {
```
Comment (QuinnDamerell) [inline scripts/octoeverywhere.sh:294]: This is great!

File: scripts/octoeverywhere.sh (hunk 1 above)
Comment (th33xitus) [reply scripts/octoeverywhere.sh:276]: Yes, calling that function handles all services beginning with "octoeverywhere".

File: scripts/octoeverywhere.sh (hunk 2 above)
Comment (th33xitus) [reply scripts/octoeverywhere.sh:279]: Oops, nope, thats something i have to fix.

File: scripts/octoeverywhere.sh (hunk 3 above)
Comment (th33xitus) [reply scripts/octoeverywhere.sh:287]: Correct, i added the evn you create during setup to the global variables and with that command i just run pip from that env and install possible new python deps into that specific env.

**Comment by QuinnDamerell** [comment]
Looks great, thanks for doing this!

**Comment by QuinnDamerell** [comment]
👏

**Pull request merged**